### ISSUE TITLE
[ERROR] Failed to fetch a distro image.

### ISSUE THREAD

**Issue by HighScorePlayer** [nullpo-head/wsl-distrod#87]
Using the distrod_wsl_launcher to install an image from linuxcontainers.org[the default choice to get the image]

### Step1: 
Run distrod_wsl_launcher.exe on win11
### Step2: 
input 2 or press [Enter], then terminate forcely
### Step3: 
Run the distrod_wsl_launcher on powershell and show the errors below
### Step4:
Network and proxy settings.

**When not using proxy:**
    0: Failed to fetch the image list.
    1: Failed to parse the distro image list of the linuxcontainer.org image server.
    2: Failed to fetch https://images.linuxcontainers.org/images/
    3:  error sending request for url (https://images.linuxcontainers.org/images/): error trying to connect ((os error 10054))

**Now using proxy:**
    0: Failed to fetch the image list.
    1: Failed to parse the distro image list of the linuxcontainer.org image server.
    2: "images/" is not available

The network are supposed to be **Okay** because I've alreaad set the UWP Lookback on and have tried to modify network security Settings of windows system. However it could not work conclusively.

**Comment by dgreenwald-ccs** [comment]
Our team facing same issue...

**Comment by HighScorePlayer** [comment]
> Same issue

It seems like the server is now under maintenance, or you can say it is updating.

**Comment by Vrashabh-Sontakke** [comment]
3 days ?

**Comment by HighScorePlayer** [comment]
> 3 days ?

I don' know. I don't even find where the notification is.

**Comment by vilcapomaparejaca1998** [comment]
Are you still getting that error?

**Comment by essn** [comment]
Same here - exact same error.

I did notice the following:

```
# distrod/libs/src/container_org_image.rs

static LINUX_CONTAINERS_ORG_BASE: &str = "https://images.linuxcontainers.org/";
```

Checking Wayback Machine it looks like https://images.linuxcontainers.org recently started redirecting to https://us.lxd.images.canonical.com/.

Wayback Machine: https://web.archive.org/web/20230801000000*/images.linuxcontainers.org.

I _think_ I successfully used Distrod with linuxcontainers.org before the Wayback Machine redirects, so it may not matter. I'll dig deeper, time allowing.

**Comment by HighScorePlayer** [comment]
> Are you still getting that error?

Yeah. I tried it just now. 
It is definitely a headache.

**Comment by HighScorePlayer** [comment]
> Same here - exact same error.
> 
> I did notice the following:
> 
> ```
> # distrod/libs/src/container_org_image.rs
> 
> static LINUX_CONTAINERS_ORG_BASE: &str = "https://images.linuxcontainers.org/";
> ```
> 
> Checking Wayback Machine it looks like https://images.linuxcontainers.org recently started redirecting to https://us.lxd.images.canonical.com/.
> 
> Wayback Machine: https://web.archive.org/web/20230801000000*/images.linuxcontainers.org.
> 
> I _think_ I successfully used Distrod with https://linuxcontainers.org before the Wayback Machine redirects, so it may not matter. I'll dig deeper, time allowing.
> 
> edit: link formatting

I guessed the domain name of the source was down or changed.

Anyway, thanks man.

**Comment by vilcapomaparejaca1998** [comment]
> > Are you still getting that error?
> 
> Yeah. I tried it just now. It is definitely a headache.

Do you know any alternative? Without considering Docker.

**Comment by yanorei32** [comment]
I use this software because WSL2 (except the store version) doesn't support systemd .
The store version has a critical issue and can not start in session 0.
https://github.com/microsoft/WSL/issues/9231
I hope this issue resolve.

**Comment by HighScorePlayer** [comment]
> > > Are you still getting that error?
> > 
> > 
> > Yeah. I tried it just now. It is definitely a headache.
> 
> Do you know any alternative? Without considering Docker.

- [Archinstall for wsl made by yuk](https://github.com/yuk7/ArchWSL), you can find an install document on the homepage.
- ArchWSL in Microsoft Store
- [LxRunOffline](https://github.com/DDoSolitary/LxRunOffline/releases) with [tutorial in Chinese](https://bbs.zsxwz.com/thread-5159.htm)

**Comment by essn** [comment]
https://github.com/DDoSolitary/LxRunOffline looks pretty cool! Glad I was following this issue. 

That said, I'd still be interested in a Distrod fix 😂

I have not tried to use Distrod recently the download issue may or may not still be a problem.

Anyone still experiencing it?

**Comment by HighScorePlayer** [comment]
> https://github.com/DDoSolitary/LxRunOffline looks pretty cool! Glad I was following this issue.
> 
> That said, I'd still be interested in a Distrod fix 😂
> 
> I have not tried to use Distrod recently the download issue may or may not still be a problem.
> 
> Anyone still experiencing it?
The problem, for me is that I can not run a desktop environment (DE) on wsl2 stablely, may be the reason in the following: 
AMD integrated graphics can not be identified by KDE. Finally I gave up installing the desktop application.

**Comment by codekoala** [comment]
I recently wiped and reinstalled everything on my laptop, and I am now encountering this issue. Do we have any solutions for distrod to actually work the way it used to?

**Comment by Vrashabh-Sontakke** [comment]
I guess the automation for fetching images is now outdated.

**Comment by CaseyLabs** [comment]
@nullpo-head - would it be possible to have a new build of `distrod` with the linuxcontainer URL fixed? Thanks!

**Comment by httpete** [comment]
> For those who found this and don't know what to do:
> 
> As for now, you can download a `rootfs.tar.xz` build artifact from LinuxContainers' public [Jenkins CI Instance](https://jenkins.linuxcontainers.org/) for the distribution of your choice.

Just to add a bit more detail - this does work, for example I navigated to:

https://jenkins.linuxcontainers.org/job/image-rockylinux/lastBuild/architecture=amd64,release=9,variant=default/

And then downloaded that rootfs.tar.xz, chose option 1, and set the path to ../rootfs.tar.xz

**Comment by httpete** [comment]
DistroD is such a great tool - I really hope someone fixes this download thing.

**Comment by CompeyDev** [comment]
> DistroD is such a great tool - I really hope someone fixes this download thing.

Yeah, it would be nice to have a functional fork. I'll see what I can do over this weekend.

**Comment by HighScorePlayer** [comment]
> > For those who found this and don't know what to do:
> > As for now, you can download a `rootfs.tar.xz` build artifact from LinuxContainers' public [Jenkins CI Instance](https://jenkins.linuxcontainers.org/) for the distribution of your choice.
> 
> Just to add a bit more detail - this does work, for example I navigated to:
> 
> https://jenkins.linuxcontainers.org/job/image-rockylinux/lastBuild/architecture=amd64,release=9,variant=default/
> 
> And then downloaded that rootfs.tar.xz, chose option 1, and set the path to ../rootfs.tar.xz

I have tried to use linux distributions with "tar.gz"-like file suffix to deploy the arch distribution directly but in the end it seams like the tool Distro with tar.xz is best.
